framework/src/application/application.ts

```diff
@@ -156,38 +159,45 @@ export class Application {
 		genesisBlock: GenesisBlockJSON,
 		config: Partial<ApplicationConfig> = {},
 	) {
-		const errors = liskValidator.validate(genesisBlockSchema, genesisBlock);
+		const errors = validator.validate(genesisBlockSchema, genesisBlock);
 		if (errors.length) {
 			throw errors;
 		}
 		this._genesisBlock = genesisBlock;
 
 		// Don't change the object parameters provided
 		// eslint-disable-next-line no-param-reassign
-		config.rootPath = config.rootPath?.replace('~', os.homedir);
-		let appConfig = _.cloneDeep(config);
+		const appConfig = _.cloneDeep(applicationConfigSchema.default);
 
 		appConfig.label =
-			appConfig.label ??
+			config.label ??
 			`lisk-${this._genesisBlock.header.transactionRoot.slice(0, 7)}`;
 
-		appConfig = configurator.getConfig(appConfig, {
-			failOnInvalidArg: process.env.NODE_ENV !== 'test',
-		}) as ApplicationConfig;
+		mergeDeep(appConfig, config);
+		appConfig.rootPath = appConfig.rootPath.replace('~', os.homedir());
+		const applicationConfigErrors = validator.validate(
+			applicationConfigSchema,
+			appConfig,
+		);
+		if (applicationConfigErrors.length) {
+			throw new Error(applicationConfigErrors.map(e => e.message).join(','));
+		}
 
-		// These constants are readonly we are loading up their default values
-		// In additional validating those values so any wrongly changed value
-		// by us can be catch on application startup
-		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-		const constants = validator.parseEnvArgAndValidate(constantsSchema, {});
+		const constantError = validator.validate(
+			constantsSchema,
+			constantsSchema.default,
```

Comment:
Why there will be an error for default values?

Reply:
That's what it was checking in the current implementation

Reply:
In old implementation, it was used `parseEnvArgAndValidate` which can change the values depending on defined env variables or command line arguments. So why we were doing validation on top of it. 

Here I don't thin its required anymore, as these are just hardcoded values we already know.